File: PHP_TP2/StudioBioskop.php
<?php
require_once __DIR__ . '/Film.php';

class StudioBioskop extends Film {
    private int $nomorStudio;
    private int $kapasitas;
    private string $tipeStudio;
    private string $fotoProduk;

    public function __construct(string $id, string $nama, int $tahunOperasional, string $judul, string $genre, int $durasi, int $nomorStudio, int $kapasitas, string $tipeStudio, string $fotoProduk = "uploads/studio_default.png") {
        parent::__construct($id, $nama, $tahunOperasional, $judul, $genre, $durasi);
        $this->nomorStudio = $nomorStudio;
        $this->kapasitas = $kapasitas;
        $this->tipeStudio = $tipeStudio;
        $this->fotoProduk = $fotoProduk;
    }

    public function getNomorStudio(): int { return $this->nomorStudio; }
    public function setNomorStudio(int $nomor): void { $this->nomorStudio = $nomor; }

    public function getKapasitas(): int { return $this->kapasitas; }
    public function setKapasitas(int $kap): void { $this->kapasitas = $kap; }

    public function getTipeStudio(): string { return $this->tipeStudio; }
    public function setTipeStudio(string $tipe): void { $this->tipeStudio = $tipe; }

    public function getFotoProduk(): string { return $this->fotoProduk; }
    public function setFotoProduk(string $foto): void { $this->fotoProduk = $foto; }
}
?>

File: CPP/StudioBioskop.cpp
#include "Film.cpp"

class StudioBioskop : public Film {
private:
    int nomorStudio;
    int kapasitas;
    string tipeStudio;

public:
    StudioBioskop() : Film() {}
    StudioBioskop(string id, string nama, int tahunOperasional, string judul, string genre, int durasi, int nomorStudio, int kapasitas, string tipeStudio)
        : Film(id, nama, tahunOperasional, judul, genre, durasi) {
        this->nomorStudio = nomorStudio;
        this->kapasitas = kapasitas;
        this->tipeStudio = tipeStudio;
    }

    // Getter & Setter
    int getNomorStudio() const { return nomorStudio; }
    void setNomorStudio(int nomorStudio) { this->nomorStudio = nomorStudio; }

    int getKapasitas() const { return kapasitas; }
    void setKapasitas(int kapasitas) { this->kapasitas = kapasitas; }

    string getTipeStudio() const { return tipeStudio; }
    void setTipeStudio(string tipeStudio) { this->tipeStudio = tipeStudio; }
};